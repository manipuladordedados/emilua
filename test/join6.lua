-- This test ensures join() propagates values back from the joinee.

local println = require('println')
local sleep_for = require('sleep_for')

local fib = spawn(function()
    println('secondary fiber starts')
    sleep_for(20)
    error('tag', 0)
end)

println('secondary fiber spawned')
sleep_for(10)
local ok, e = pcall(function() return fib:join() end)
println(tostring(ok))
println(e)
println('end of main fiber')
