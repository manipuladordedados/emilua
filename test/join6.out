secondary fiber spawned
secondary fiber starts
false
tag
end of main fiber
